models/currency.js

```diff
@@ -0,0 +1,188 @@
+const firestore = require('../utils/firestore')
+const walletModel = firestore.collection('wallets')
+const bankModel = firestore.collection('banks')
+const currencyExchangeModel = firestore.collection('currencyExchange')
+const transactionModel = firestore.collection('transactions')
+const { extractReferenceDocumentData, extractReferenceDocumentId } = require('../utils/firestore-helper')
+
+const allCurrencyName = 'allCurrencies'
+
+/**
+ * Fetch the currency exchange rates
+ * @return {Promise<exchangeData>}
+ */
+const fetchExchangeRates = async () => {
+  try {
+    const exchangeSnapshot = await currencyExchangeModel.doc(allCurrencyName).get()
+    if (exchangeSnapshot.exists) {
+      return exchangeSnapshot.data()
+    }
+    return undefined
+  } catch (err) {
+    logger.error('Error while getting currency rates from exchange', err)
+    throw err
+  }
+}
+
+/**
+ * Fetch the currency exchange from the bank
+ * @return {Promise<exchangeData>}
+ */
+const fetchCurrencyAvailablity = async (bankId) => {
+  try {
+    const bankSnapshot = await bankModel.where('bankId', '==', bankId).limit(1).get()
+    if (!bankSnapshot.empty) {
+      const bank = (bankSnapshot.docs[0] && (bankSnapshot.docs[0].data()))
+      if (bank.isActive) {
+        return bank.currency
+      }
+    }
+    return undefined
+  } catch (err) {
+    logger.error('Error while getting currency data from bank', err)
+    throw err
+  }
+}
+
+/**
+ * Sets the currency exchange rates
+ * @return {Promise<exchangeData>}
+ */
+const createCurrencyRates = async (currencyData) => {
+  try {
+    await currencyExchangeModel.add(currencyData)
+    await currencyExchangeModel.doc(allCurrencyName).set({
+      [currencyData.src]: {
+        [currencyData.target]: currencyData.quantity
+      },
+      lastUpdate: Date.now()
+    }, {
+      merge: true
+    })
+    return true
+  } catch (err) {
+    logger.error('Error while adding currency rates into exchange', err)
+    throw err
+  }
+}
+
+/**
+ * Fetch the all banks availabe name and id
+ * @return {Promise<exchangeData>}
+ */
+const getAllBanks = async () => {
+  try {
+    const bankSnapshot = await bankModel.get()
+    const bankData = []
+    bankSnapshot.forEach(bank => {
+      const data = bank.data()
+      if (data.isActive) {
+        bankData.push({ bankId: data.bankId, bankName: data.bankName })
```

Comment:
Could we destructure `bankId` and `bankName` and use? Wouldn't it increase the readability? What do you think?